### PR TITLE
Add new prop to _XpuDevicePropertie for triton gemm optimization

### PR DESCRIPTION
**Pull request by guangyey**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #131738

# Motivation
This PR aims to add new properties to `_XpuDevicePropertie` for triton gemm optimization.

# Additional Context
`ext_oneapi_supports_cl_extension` is not a ABI-neutral API. It depends on compiler 2025.0. For more details, see https://github.com/intel/llvm/pull/13212

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by gujinghui — APPROVED** [review]

File: c10/xpu/XPUFunctions.cpp [hunk 1]
```diff
@@ -89,6 +89,10 @@ void initDeviceProperties(DeviceProp* device_prop, int device) {
 #define ASSIGN_DEVICE_ASPECT(member) \
   device_prop->has_##member = raw_device.has(sycl::aspect::member);
 
+#define ASSIGN_EXP_CL_ASPECT(member) \
+  device_prop->has_##member =        \
+      raw_device.ext_oneapi_supports_cl_extension("cl_intel_" #member);
```
Comment (ZzEeKkAa) [inline c10/xpu/XPUFunctions.cpp:94]:
```suggestion
  sycl::ext::oneapi::experimental::cl_version version_##member;            \
  device_prop->has_##member = raw_device.ext_oneapi_supports_cl_extension( \
      "cl_intel_" #member, &version_##member);
```

The compilation does not fail, however I'm getting core dump without second argument:
```bash
$ python -c "import torch; print(torch.xpu.get_device_capability())"
[1]    130676 segmentation fault (core dumped)  python -c "import torch; print(torch.xpu.get_device_capability())"
```

Comment (guangyey) [reply c10/xpu/XPUFunctions.cpp:94]: Thanks for your suggestion. I will check it.

Comment (ZzEeKkAa) [reply c10/xpu/XPUFunctions.cpp:94]: I'm using PTDB 0.5.2.18 with updated ubuntu 22.04 and python 3.9 venv installed from deadsnakes/ppa. Just reproduced the issue.

```
icpx --version
Intel(R) oneAPI DPC++/C++ Compiler 2024.1.3 (2024.1.3.20240604)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /opt/intel/oneapi/compiler/2024.1/bin/compiler
Configuration file: /opt/intel/oneapi/compiler/2024.1/bin/compiler/../icpx.cfg
```

Comment (guangyey) [reply c10/xpu/XPUFunctions.cpp:94]: OK, thanks. The compiler team has confirmed this is a bug.

Comment (guangyey) [reply c10/xpu/XPUFunctions.cpp]: I will merge this PR once our CI upgrades to the appropriate compiler.

Comment (ZzEeKkAa) [reply c10/xpu/XPUFunctions.cpp]: Thank you!

Comment (guangyey) [reply c10/xpu/XPUFunctions.cpp]: No problem!

**Comment by guangyey** [comment]
@ZzEeKkAa bundle 0.5.3 has been released. Let us merge this PR.

**Pull request closed without merging**